### PR TITLE
Feature/drawer

### PR DESCRIPTION
**Pull request by TahaNamdar**

### CODE REVIEW

File: src/components/drawer/content/Drawer.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  OverlayTop,
+  OverlayBottom,
+  OverlayContentTopBottom,
+} from '../styles/Drawer.styles';
+
+const Drawer: React.FC<DrawerProps> = ({ children, ...rest }: DrawerProps) => {
+  const isMount = useIsMount();
+
+  React.useEffect(() => {
+    if (isMount) {
```
Comment (alirezatahani) [inline src/components/drawer/content/Drawer.tsx:16]: change it to 
if ( !isMounted && !rest.visible )

File: src/components/drawer/styles/Drawer.styles.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  })
+);
+export const OverlayContentLeftRight = styled.div(({ theme }) => ({
+  flex: 'none',
+  textAlign: 'center',
+  padding: 20,
+  color: theme.palette.common.white,
+}));
+
+export const OverlayTop = styled.div<DrawerProps>(
```
Comment (alirezatahani) [inline src/components/drawer/styles/Drawer.styles.ts:24]: OverlayTop && OverlayBottom && OverlayLeftRight 

they can be one component

Comment (TahaNamdar) [reply src/components/drawer/styles/Drawer.styles.ts:24]: nemishe bayad behet begam chera to whats app

File: src/components/drawer/styles/Drawer.styles.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
+import styled from 'styled-components';
+import { DrawerProps } from '../content/drawer_types';
+import { Position } from '../utils/constants';
+export const OverlayLeftRight = styled.div<DrawerProps>(
+  ({ theme, visible, position, width, height }) => ({
+    backgroundColor: theme.palette.secondary.main,
+    position: 'absolute',
```
Comment (alirezatahani0) [inline src/components/drawer/styles/Drawer.styles.ts:9]: chera absolute ?

File: src/components/drawer/styles/Drawer.styles.ts [hunk 4]
```diff
@@ -0,0 +1,61 @@
+import styled from 'styled-components';
+import { DrawerProps } from '../content/drawer_types';
+import { Position } from '../utils/constants';
```
Comment (alirezatahani0) [inline src/components/drawer/styles/Drawer.styles.ts:3]: need empty space after imports

File: src/components/drawer/styles/Drawer.styles.ts [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+);
+export const OverlayBottom = styled.div<DrawerProps>(
+  ({ theme, visible, position, height }) => ({
+    backgroundColor: theme.palette.secondary.main,
+    position: 'absolute',
+    overflow: 'hidden',
+    borderRadius: '0px 2px 4px 0px',
+    zIndex: 1,
+    width: visible ? '100%' : 0,
+    ...Position[position],
```
Comment (alirezatahani0) [inline src/components/drawer/styles/Drawer.styles.ts]: onaei ke spread mishan ro biar avval benevis

File: src/components/drawer/styles/Drawer.styles.ts [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+    transform: visible ? 'translateY(0px)' : 'translateY(-500px)',
+    transition: 'transform 1s ease-in-out',
+    height: visible ? `${height}px` : 0,
+  })
+);
+export const OverlayBottom = styled.div<DrawerProps>(
+  ({ theme, visible, position, height }) => ({
+    backgroundColor: theme.palette.secondary.main,
+    position: 'absolute',
```
Comment (alirezatahani0) [inline src/components/drawer/styles/Drawer.styles.ts:43]: chera absolute ??

**Review by alirezatahani0 — CHANGES_REQUESTED** [review]

File: src/components/drawer/styles/Drawer.styles.ts (hunk 3 above)
Comment (TahaNamdar) [reply src/components/drawer/styles/Drawer.styles.ts:9]: chun overlay daqiqan az bala chasbide be goshe baz she bar daram kharab mishe

**Pull request closed without merging**